components/edit_post/edit_post.tsx

```diff
@@ -0,0 +1,494 @@
+import {isMac} from 'utils/utils';
+import * as Utils from 'utils/utils';
+
+import DeletePostModal from 'components/delete_post_modal';
+import EmojiPickerOverlay from 'components/emoji_picker/emoji_picker_overlay';
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import Textbox, {TextboxClass} from 'components/textbox';
+import EmojiIcon from 'components/widgets/icons/emoji_icon';
+import {isFirefox} from '../../utils/user_agent';
+
+type OpenModal = {
+    ModalId: string;
+    dialogType: typeof React.Component;
+    dialogProps: {
+        post: Post;
+        isRHS?: boolean;
+    };
+};
+
+export type Actions = {
+    addMessageIntoHistory: (message: string) => void;
+    editPost: (input: Partial<Post>) => Promise<Post>;
+    unsetEditingPost: () => void;
+    openModal: (input: OpenModal) => void;
+    scrollPostListToBottom: () => void;
+}
+
+export type Props = {
+    canEditPost?: boolean;
+    canDeletePost?: boolean;
+    readOnlyChannel?: boolean;
+    channelId: string;
+    codeBlockOnCtrlEnter: boolean;
+    ctrlSend: boolean;
+    config: {
+        EnableEmojiPicker?: string;
+        EnableGifPicker?: string;
+    };
+    maxPostSize: number;
+    useChannelMentions: boolean;
+    editingPost: {
+        post: Post | null;
+        postId?: string;
+        refocusId?: string;
+        title?: string;
+        isRHS?: boolean;
+    };
+    actions: Actions;
+};
+
+export type State = {
+    editText: string;
+    caretPosition: number;
+    postError: React.ReactNode;
+    errorClass: string | null;
+    showEmojiPicker: boolean;
+    renderScrollbar: boolean;
+    scrollbarWidth: number;
+    prevShowState: boolean;
+};
+
+const {KeyCodes} = Constants;
+
+const TOP_OFFSET = 0;
+const RIGHT_OFFSET = 10;
+
+const EditPost = ({editingPost, actions, ...rest}: Props): JSX.Element | null => {
+    const [editText, setEditText] = useState<string>(
+        editingPost.post?.message_source || editingPost.post?.message || '',
+    );
+    const [caretPosition, setCaretPosition] = useState<number>(editText.length);
+    const [postError, setPostError] = useState<React.ReactNode | null>(null);
+    const [errorClass, setErrorClass] = useState<string>('');
+    const [showEmojiPicker, setShowEmojiPicker] = useState<boolean>(false);
+    const [renderScrollbar, setRenderScrollbar] = useState<boolean>(false);
+
+    const textboxRef = useRef<TextboxClass>(null);
+    const emojiButtonRef = useRef<HTMLButtonElement>(null);
+    const wrapperRef = useRef<HTMLDivElement>(null);
+
+    const {formatMessage} = useIntl();
+
+    const shouldScroll = (elementRect: DOMRect, containerRect: DOMRect): boolean => {
+        if (!elementRect || !containerRect) {
+            return false;
+        }
+
+        return elementRect.top < containerRect.top || elementRect.bottom > containerRect.bottom;
+    };
+
+    useEffect(() => {
+        const className = `post-list__dynamic${editingPost.isRHS ? '--RHS' : ''}`;
```

Comment:
I originally added it because the container was not scrolled into view correctly. I just tested it again using `setTimeout` and now it works. Thanks for catching that ... it makes the whole process a bit better.